autobotAI_integrations/utils/aws_mcp_auth.py

```
@@ -0,0 +1,187 @@
+AWS MCP Server authentication helpers.
+
+The official AWS MCP Server (https://aws-mcp.{region}.api.aws/mcp) requires
+IAM SigV4 signing on HTTP requests. These helpers build MCPCreds and sign
+requests without putting long-lived secrets in HTTP header maps when IAM
+fields are used instead.
+"""
+
+from __future__ import annotations
+
+import re
+from typing import Dict, Optional, Tuple
```

Ruff (0.15.18)

[warning] `typing.Dict` is deprecated, use `dict` instead

(UP035)